testing/test_pytester.py

```diff
@@ -776,3 +776,28 @@ def test_makefile_joins_absolute_path(testdir: Testdir) -> None:
     else:
         p1 = testdir.makepyfile(**{str(absfile): ""})
         assert str(p1) == (testdir.tmpdir / absfile) + ".py"
+
+
+def test_testdir_created_files_none(testdir):
+    assert set() == testdir.created_files
+
+
+def test_testdir_duplicate_paths(testdir):
+    testdir.makepyfile("foo")
+    testdir.makepyfile("bar")
+    assert len(testdir.created_files) == 1
+
+
+def test_files_by_name_are_correct(testdir):
+    testdir.maketxtfile("foo")
+    testdir.maketxtfile(custom="foobar")
+    assert len(testdir.created_files) == 2
+
+
+def test_testdir_create_file_of_type(testdir):
+    data = "hello world"
+    one = testdir.makepyfile(data)
+    two = testdir.maketxtfile(data)
+    three = testdir.makeconftest(data)
+    four = testdir.makeini(data)
+    assert {one, two, three, four} == testdir.created_files
```

Comment:
```suggestion
    assert testdir.created_files == {one, two, three, four}
```

For consistency with the rest of the tests: "obtained == expected"